setup.py

```diff
@@ -52,9 +52,12 @@ def read_version():
 
 gethostname = setuptools.Extension(
     "gethostname",
-    sources=["src/sagemaker_training/c/gethostname.c", "src/sagemaker_training/c/jsmn.c"],
+    sources=[
+        "src/sagemaker_training/c/gethostname.c",
+        "src/sagemaker_training/c/jsmn.c",
+    ],
     include_dirs=["src/sagemaker_training/c"],
-    extra_compile_args=["-Wall", "-shared", "-export-dynamic", "-ldl"],
+    extra_compile_args=["-Wall", "-shared", "-Wl,-export-dynamic", "-ldl"],
```

Comment:
nit: separate out `-Wl` and `-export-dynamic`.

Reply:
Suggestion: explore separating out linker and compile args if any, guideline- https://setuptools.pypa.io/en/latest/userguide/ext_modules.html#compiler-and-linker-options

Reply:
I don't understand your suggestion. The _-Wl_ option is a comma separated list of options to forward to the linker. In the link you suggested there is an example where -Wl is joint with the option "-rpath"

> then, a linker-specific option like _-Wl,-rpath_ for each element of Extension.runtime_library_dirs,

I did some more tests, `gcc` accepts the stand-alone -export-dynamic but clang does not, it requires it being passed with "-Wl" to properly forward it to the linker. Additionally, separating "-Wl" and "-export-dynamic" breaks in both `gcc` and `clang` since "-Wl" alone is an unrecognized option. 

If you want I could move the option to the argument _runtime_library_dirs_ or _extra_link_args_ but the value would still be `"-Wl,-export-dynamic"`. The only difference it would make is the order in which the option will be appended into the command to invoke `gcc` or `clang`. I'm ok with either change.